src/engine/PathSearch.cpp

```diff
@@ -357,9 +360,21 @@ PathsLimited PathSearch::findPaths(const Id& source,
       currentPath.pop_back();
     }
 
+    auto edgeEnd = edge.end_.getBits();
+    if (numPathsPerTarget) {
+      numPathsPerNode.try_emplace(edgeEnd, 0);
```

Comment:
this computes the hash three times. A cheaper way is

```
auto [it, _] = try_emplace... (like you did before)
auto& numPaths = it->second;
// directly read and write to numPaths from here on.
```
or much simpler:

```
auto numPaths = ++numPathsPerNode[edgeEnd];
if (numPaths too large) {
  continue;
 }
```
The second one might theoretically overlow, but if 64 bits overflow then we have other issues:)